test/new-e2e/pkg/utils/e2e/client/ecs/session-manager-plugin.go

```diff
@@ -0,0 +1,118 @@
+	"github.com/aws/session-manager-plugin/src/log"
+	"github.com/aws/session-manager-plugin/src/message"
+	"github.com/aws/session-manager-plugin/src/retry"
+	"github.com/aws/session-manager-plugin/src/sessionmanagerplugin/session"
+	"github.com/google/uuid"
+)
+
+// Slightly modified version of OpenDataChannel function in github.com/aws/session-manager-plugin/src/sessionmanagerplugin/session/sessionhandler.go
+func openDataChannel(s *session.Session, logger log.T, stopChan chan bool) (err error) {
+	retryParams := retry.RepeatableExponentialRetryer{
+		GeometricRatio:      config.RetryBase,
+		InitialDelayInMilli: rand.Intn(config.DataChannelRetryInitialDelayMillis) + config.DataChannelRetryInitialDelayMillis,
+		MaxDelayInMilli:     config.DataChannelRetryMaxIntervalMillis,
+		MaxAttempts:         config.DataChannelNumMaxRetries,
+	}
+
+	s.DataChannel.Initialize(logger, s.ClientId, s.SessionId, s.TargetId, s.IsAwsCliUpgradeNeeded)
+	s.DataChannel.SetWebsocket(logger, s.StreamUrl, s.TokenValue)
+	s.DataChannel.GetWsChannel().SetOnMessage(
+		func(input []byte) {
+			s.DataChannel.OutputMessageHandler(logger, func() { stopChan <- true }, s.SessionId, input)
+		})
+	s.DataChannel.RegisterOutputStreamHandler(s.ProcessFirstMessage, false)
+
+	if err = s.DataChannel.Open(logger); err != nil {
+		logger.Errorf("Retrying connection for data channel id: %s failed with error: %s", s.SessionId, err)
+		retryParams.CallableFunc = func() (err error) { return s.DataChannel.Reconnect(logger) }
+		if err = retryParams.Call(); err != nil {
+			logger.Error(err)
+		}
+	}
+
+	s.DataChannel.GetWsChannel().SetOnError(
+		func(_ error) {
+			logger.Errorf("Trying to reconnect the session: %v with seq num: %d", s.StreamUrl, s.DataChannel.GetStreamDataSequenceNumber())
+			retryParams.CallableFunc = func() (err error) { return s.ResumeSessionHandler(logger) }
+			if err = retryParams.Call(); err != nil {
+				logger.Error(err)
+			}
+		})
+
+	// Scheduler for resending of data
+	s.DataChannel.ResendStreamDataMessageScheduler(logger)
+
+	return nil
+}
+
+// Slightly modified version of Execute function in github.com/aws/session-manager-plugin/src/sessionmanagerplugin/session/session.go
+func execute(s *session.Session, logger log.T) (string, error) {
+	stopChannel := make(chan bool)
+	var payload []byte
+	if err := openDataChannel(s, logger, stopChannel); err != nil {
+		logger.Errorf("Error in Opening data channel: %v", err)
+		return "", err
+	}
+	s.DataChannel.RegisterOutputStreamHandler(func(logger log.T, msg message.ClientMessage) (bool, error) {
+		payload = append(payload, msg.Payload...)
+		return true, nil
+	}, true)
+
+	select {
+	case <-s.DataChannel.IsSessionTypeSet():
+		if <-stopChannel {
+			return string(payload), nil
+		}
+	case <-stopChannel:
+		return "", fmt.Errorf("Failed to initialize session")
+	}
+	return "", nil
+}
+
+func retrieveResultFromExecOutput(c *Client, output *ecs.ExecuteCommandOutput, task, container string) (string, error) {
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities. -->
**💭 thought**
As this is the most cryptic part of the code, could help to have a comment about what this is doing